static/app/views/dashboardsV2/create.tsx

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import {browserHistory, PlainRoute, WithRouterProps} from 'react-router';
+import styled from '@emotion/styled';
+
+import {createDashboard} from 'app/actionCreators/dashboards';
+import {addSuccessMessage} from 'app/actionCreators/indicator';
+import {Client} from 'app/api';
+import Feature from 'app/components/acl/feature';
+import Alert from 'app/components/alert';
+import LightWeightNoProjectMessage from 'app/components/lightWeightNoProjectMessage';
+import GlobalSelectionHeader from 'app/components/organizations/globalSelectionHeader';
+import SentryDocumentTitle from 'app/components/sentryDocumentTitle';
+import {t} from 'app/locale';
+import {PageContent} from 'app/styles/organization';
+import space from 'app/styles/space';
+import {Organization} from 'app/types';
+import {trackAnalyticsEvent} from 'app/utils/analytics';
+import withApi from 'app/utils/withApi';
+import withOrganization from 'app/utils/withOrganization';
+
+import Controls from './controls';
+import Dashboard from './dashboard';
+import {DEFAULT_STATS_PERIOD, EMPTY_DASHBOARD} from './data';
+import DashboardTitle from './title';
+import {DashboardDetails, DashboardState, Widget} from './types';
+import {cloneDashboard} from './utils';
+
+const UNSAVED_MESSAGE = t('You have unsaved changes, are you sure you want to leave?');
+
+type Props = {
+  api: Client;
+  organization: Organization;
+  route: PlainRoute;
+} & WithRouterProps<{orgId: string; dashboardId: string}, {}>;
+
+type State = {
+  dashboard: DashboardDetails;
+  dashboardState: DashboardState;
+};
+
+class CreateDashboard extends React.Component<Props, State> {
```

Comment:
I haven't hooked up the 'Create Dashboard' button from the detail page here and I haven't pulled in any of the metric widget code into this component.

Reply:
I believe that we should also have here the codes that lead to the widget builder, since we plan to soon replace the widget creation dialog with the widget builder

Reply:
I know ahh, but there has to be a better way to do this. It's going to be rough maintaining duplicates of the edit and create states in two separate files- maybe we can fix that somehow by pulling all of the functions that handle widget builder stuff into a common utils file. But then, I also need to figure out how to resolve the references to dashboard id in all the widget builder urls (since "create dashbaords" won't have an ID), so there will have to be a refactor there as well.